### DISCUSSION TITLE
Mobile Only x Responsive

### DISCUSSION THREAD

**Discussion by rubenmarcus** [frontendbr/forum#95] [Perguntas e Respostas]
Qual técnica vocês preferem? 

O que é melhor em questão de manutenção e  performance?

**Comment by bernardodiasc** [comment]
Pode elaborar melhor sobre mobile only? É uma versão separada pra mobile ou um site "mobile only" mesmo?

**Comment by rubenmarcus** [comment]
Versão separada..  pode ser detectado via servidor ou js mesmo

**Comment by felipefialho** [comment]
Cara, sou a favor de aplicativos responsivos em 92% dos casos (um número inventado por mim, rs).

De verdade acho improvável que a grande maioria dos produtos que usam app ou site especifico realmente precisavam, hoje é possível resolver quase tudo com HTML5 e técnicas que estamos carecas de saber.

Claro que não estou falando de aplicações como Uber, Instagram, etc, etc, que obviamente necessitam de um app separado.

Também não estou falando de projetos de websites antigos de grandes portais, que acabam criando uma versão mobile como medida temerária, por ser quase impossível adaptar um projeto antigo para responsivo.

**Comment by rubenmarcus** [comment]
sim mas a questão em performance.. pq o responsivo se vc pegar um framework como o bootstrap ele le tudo duma vez, seja o css, js etc, gastando recurso que não necessariamente ia precisar pra aquela versão......

o ideal nao seria fazer um site em que o js e o css carregasse pra cada  tipo de tela?e não pra todas e  o browser responde-se a isso?

**Comment by jhonnymoreira** [comment]
@rubenmarcus essa foi uma dúvida que tive enquanto estava trabalhando no Gaveteiro. E cara, pra ser honesto, mesmo tendo apoiado o domínio separado e etc durante um bom tempo, hoje não consigo ver tanto lucro assim. Se tu tá lidando com um framework, o ideal que tu deve fazer é passar o framework pra um pré-processador e a partir disso selecionar o que vai pra produćão e será realmente utilizado (isso levando em consideraćão o CSS). A intenćão, independente da plataforma, é deixar o teu site o mais performático e usável possível, novamente, independente de plataforma.

O que tenho medo nessa abordagem de divisão de domínio pra mobile e desktop é a brecha que fica aberta pra fazer gambiarra e justificar que pra um róla fazer isso e pro outro não, e também a manutenćão dobrada, já que seria manutenćão pra cada domínio (imaginando que cada um teria seu stylesheet e etc...).

**Comment by lagden** [comment]
@rubenmarcus

# Vejamos os tops

## Facebook (Separado)

[Facebook Desktop](https://www.facebook.com/) e [Facebook Mobile](https://m.facebook.com/?hrc=1&_rdr)

Observe que o Facebook preferiu a linha do separado, pois na versão desktop é uma poluição visual do capeta!!! Mas eles exploram bem e bastante as propagandas nesse tipo de formato. Já emendam um chat, os games, etc...

<img width="1440" alt="fdesk" src="https://cloud.githubusercontent.com/assets/130963/14450445/cf15cbae-0054-11e6-808a-97abd49a9f81.png">

A versão mobile é bem clean e o javascript deve ser mais light, presumo eu!
Mas essa versão passa uma sensação de que foi feito apenas pra quebrar o galho, pois fazer um responsivo ia dar trabalho!

<img width="610" alt="fmobile" src="https://cloud.githubusercontent.com/assets/130963/14450446/cf19df00-0054-11e6-8af5-803f693920de.png">

---

## Google Plus (Responsivo)

[Google Plus](https://plus.google.com/)

Google Plus é responsivo e já segue o linha do mobile style...

<img width="1196" alt="gplus" src="https://cloud.githubusercontent.com/assets/130963/14450279/63a913c2-0053-11e6-9904-4eaaf9582e8f.png">

O Google está pensando no `User Experience (UX)` total!!!
Dessa forma o usuário dele estará confortável tanto no desktop quanto no mobile. Além do `User Interface (UI)` ser a mesma em todos os dispositivos

---

### Minha conclusão

Eu gosto da linha de pensamento do `Google Plus` devido ao fato das pessoal estarem mais no mobile do que no desktop (linha do mobile style).
Assim, quando o usuário do `Google Plus` abrir o site no seu desktop ou no mobile, ele se sentirá em "casa".

**Comment by bernardodiasc** [comment]
Particularmente prefiro o modelo do Facebook, mas isso é apenas opinião pessoal. O que vejo é que o Google Plus é bem legal no mobile, mas no desktop aparenta ser um mobile grandão, nunca curti mosaico de texto em streams (funciona bem num jornal de papel ou até um site de fotografia/videos). Já o Facebook entrega o necessário no mobile e uma aplicação completa no desktop, além disso, a versão desktop é responsiva também para telas menores tudo encaixa bem, muitos extras só aparecem quando tem espaço sobrando.

**Comment by lagden** [comment]
> a versão desktop é responsiva...

sqn!!!

**Comment by bernardodiasc** [comment]
é sim. nao to falando de desktop<->mobile, to falando de ajustar em desktop com telas menores. a coluna lateral com joguinhos some, o chat vira uma abinha retratil, etc.

**Comment by lagden** [comment]
@bernardodiasc relaxa!!
O Github é outro exemplo de site separado...
Quebra um galho mas ainda fica devendo algumas features que poderiam ter no mobile. exemplo: editar uma issue!

**Comment by bernardodiasc** [comment]
> @bernardodiasc relaxa!!

hehehehe to de boa, só argumentando de leve, aqui nao é o facebook pra tacar fogo em tudo kkkkk, só um pouquinho de lenha na fogueira mesmo. Nao tenho opiniao sobre qual metodo é melhor, mas quanto mais informaçao que somarmos na conversa, melhora nossa capacidade de oferecer boas solucoes no trampo. nao precisamos concordar o tempo todo :)

acho que existem estratégias que as empresas (as que tem recursos, tanto tecnicos quanto financeiros, pra escolher o que quiser) escolherem para seus produtos que nao é visivel do ponto de vista de fora. tento entender isso. Pq o Github nao é responsivo? _imagino_ que seja pq é uma ferramenta para desktop, a experiencia no mobile nao é compativel, revisar PR no celular é horrivel e nunca vai ser bom (opiniao pessoal), entao no mobile pode focar nas issues sendo uma plataforma diferente... sei lá, realmente nao sei, só fazendo suposiçoes.

**Comment by felipefialho** [comment]
Eu sinto falta de uma versão responsiva para o Github, muitas features são deixadas de fora no mobile e na minha opinião não tinha necessidade.

Claro que não pretendo revisar pull-request nem nada muito pesado pelo celular, mas queria no mínimo ter uma navegação mais consistente.

**Comment by lagden** [comment]
O que dá para perceber é:

Se o web app é altamente complexo, a empresa resolve criar uma versão mobile com as `features` reduzidas para não deixar o usuário na mão!!

Já o responsivo, a tecnologia é suportada sem problemas, ou seja, não existe a necessidade de fazer uma separação, basta dar um tapa no design...
